### PR TITLE
feature: add latency to delayed job metrics

### PR DESCRIPTION
**Pull request by martinramirez7**
This PR adds latency for delayed job jobs

### CODE REVIEW

**Comment by martinramirez7** [comment]
Hey @SamSaffron, would you take a quick look of this please? thanks in advance!

**Comment by SamSaffron** [comment]
can you also amend the readme here? Is the latency going to be accurate, can you explain the value of the metric? Also looks like CI is failing

**Comment by martinramirez7** [comment]
we are trying to implement queues based on latency, the whole idea is explained [here](https://github.com/dmagliola/happy_queues). The main concept is to have queues that ensures that all of their jobs will start running in no less time than certain latency. This metric allows to detect if an increase in CPU or number of workers  is needed for a certain queue.

> can you also amend the readme here?

where exactly?

**Comment by SamSaffron** [comment]
> where exactly?

REAMDE.md in the root :)

**Comment by martinramirez7** [comment]
I already included the metric in the readme I'm not sure if you wanted me to add anything else. About the accuracy, there might be some loss of precision because of the sleep_delay parameter implemented by delayed job I am mentioning it in the README too.
Also, all the workflows should be passing now, would you please run the tests again?

**Comment by SamSaffron** [comment]
I think we had some internal examples of time mocking without timecop, can we move to that, after that is done I am happy to merge!

**Comment by martinramirez7** [comment]
Thanks Sam! Timecop removed

**Review by SamSaffron — APPROVED** [review]

**Comment by SamSaffron** [comment]
looks good, thanks!

File: lib/prometheus_exporter/instrumentation/delayed_job.rb [hunk 1]
```diff
@@ -31,6 +31,7 @@ def initialize(client: nil)
     def call(job, max_attempts, enqueued_count, pending_count, *args, &block)
       success = false
       start = ::Process.clock_gettime(::Process::CLOCK_MONOTONIC)
+      latency = Time.current - job.run_at
```
Comment (sosedoff) [inline lib/prometheus_exporter/instrumentation/delayed_job.rb:34]: Is it safe to call `Time.current` here because delayed job depends on active support?

Comment (martinramirez7) [reply lib/prometheus_exporter/instrumentation/delayed_job.rb:34]: Exactly, they both handle the time in the same way, DJ uses `Time.zone` within `db_time_now` method for setting the `run_at` value.

**Pull request merged**